lib/derivations.nix

```diff
@@ -206,4 +206,40 @@ in
   optionalDrvAttr =
     cond:
     value: if cond then value else null;
+
+  /**
+    Wrap a derivation such that instantiating it produces a warning.
+
+    # Inputs
+
+    `msg`
+    : The warning message to emit (via lib.warn).
+
+    `pkg`
+    : The derivation to wrap.
+
+    # Examples
+    :::{.example}
+    ## `lib.derivations.wrapWithWarn` usage example
+
+    ```nix
+    {
+      myPackage = wrapWithWarn "myPackage has been renamed to my-package" my-package;
+    }
+    ```
+
+    :::
+  */
+  wrapWithWarn =
```

Comment:
I'm ok for this to be top level in `lib`, but then a slightly more specific name would be nice.
Starting with `warn` makes it more discoverable in the repl.
How about `warnOnInstantiate` or `warnOutputs`?
(well, depends on the outcome of the other comment - maybe skip this one for now)

Reply:
I've switched to `warnOnInstantiate` at least for now, I agree that starting with `warn` is nice.